source/textInfos/offsets.py

```diff
@@ -208,7 +209,10 @@ def _get_boundingRects(self):
 			pass
 		# If the inclusive end offset is greater than the start offset, we are working with a range.
 		# If not, i.e. the range only contains one character, we have only one location to deal with.
-		objLocation = self.obj.rootNVDAObject.location if isinstance(self.obj, TreeInterceptor) else self.obj.location
+		obj = self.obj.rootNVDAObject if isinstance(self.obj, TreeInterceptor) else self.obj
+		objLocation = obj.location or api.getForegroundObject().location

```

Comment:
Would an idea be to walk up the parents until finding a valid location? Foreground does seem a little inaccurate to me, when a direct parent would be much closer to the correct coordinates? Do make sure to break the loop after say 100 iterations just in case  there is a broken parent chain that repeetes.  However, if this idea doesn't really suit what you're going for, I'm happy to keep  the foreground idea.

Reply:
This makes sense!